lib/utils.ts

```diff
@@ -14,12 +14,16 @@ export const objectToString = (obj: unknown) =>
   Object.prototype.toString.call(obj)
 
 /** Safely converts any value to string, using the value's own `toString` when available. */
-export const safeToString = (val: unknown) => {
-  // Ideally, we'd just use String() for everything, but it breaks if `toString` is missing (mostly
-  // values with no prototype), so we have to use Object#toString as a fallback.
+export const safeToString = (val: unknown): string => {
+  // Using .toString() fails for null/undefined and implicit conversion (val + "") fails for symbols
+  // and objects with null prototype
   if (val === undefined || val === null || typeof val.toString === 'function') {
     return String(val)
+  } else if (Array.isArray(val)) {
```

Comment:
In the `Array` case, this 2nd conditional will never be reached because of the 1st conditional which includes `typeof val.toString === 'function'` which is true for `Array` instances.  I would suggest swapping the order here so the `isArray` check happens first.